### PR TITLE
Ports Powerfist

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Ports https://github.com/tgstation/tgstation/pull/19013

The Power first.

Syndicate traitor item that is a melee weapon that uses gas for "ammo". Simply insert a tank (emergency oxygen won't work) then go punch someone in the face. You can adjust how powerful the punch is with a wrench (more power = further throw+more damage, but more gas used up, therefore less uses) and can eject the tank mounted on it with a screwdriver.

It's horrendous at crowd combat, but it's pretty devastating one on one.

Mandatory demonstration:

img

:cl: Fox McCloud
add: Implements the powerfist and adds it to the uplink for 8 TC
/:cl:

### CODE REVIEW

**Comment by Lady-Luck** [comment]
I love the power fist. it's so bad.

**Comment by Spacemanspark** [comment]
I've always wanted the power to send someone flying across the room like that, like the singulo hammer.
:+1:

**Comment by Earthdivine** [comment]
Now add Power Armor/fix the RIG's.

**Comment by KasparoVy** [comment]
LOL FUCK
Hit a guy and it's like "PEACE" 'cause he's GONE

WOW

**Comment by IcyV** [comment]
>fix the RIG's.

Calm down there Satan

**Comment by TheDZD** [comment]
> > fix the RIG's.

> Calm down there Satan

This, very much.

**Comment by Earthdivine** [comment]
But I wanna stomp around in Power Armor Power Fisting things :C

**Comment by taukausanake** [comment]
How powerful can this thing get? Can it One-Punch Man people? Does it only work on mobs and basically not lesser hulk?

**Comment by Fox-McCloud** [comment]
36 force is the max it can do.

**Comment by taukausanake** [comment]
How much force does it take to gib someone? Can pure force gib anyone in the first place? Can it One-Punch Man people? I feel it needs to be able to do that. Think of all the !!FUN!! that can be had

**Comment by Spacemanspark** [comment]
36 force = 36 brute damage per hit.
It cannot 1 hit KO someone.

**Comment by taukausanake** [comment]
I wonder how much var editing would be needed to make 200+ force... Surely not the force var itself. You'll need to edit the pressure too I'd imagine

**Comment by AceMcLazer** [comment]
I always hoped we'd get a traitor boxing gloves. like these sort of. but when they play this

https://www.youtube.com/watch?v=TvvTacquttk

**Comment by Vivalas** [comment]
ROCKET PUNCH

**Comment by taukausanake** [comment]
@Vivalas out of curiosity where do you know that from?

**Comment by Vivalas** [comment]
MGS. Sure, it's not a flying fist you can control, but that's what it reminded me of.

File: code/game/objects/items/weapons/powerfist.dm [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+			if(3)
+				fisto_setting = 1
+		playsound(loc, 'sound/items/Ratchet.ogg', 50, 1)
+		to_chat(user, "<span class='notice'>You tweak \the [src]'s piston valve to [fisto_setting].</span>")
+	else if(istype(W, /obj/item/weapon/screwdriver))
+		if(tank)
+			updateTank(tank, 1, user)
+
+
+/obj/item/weapon/melee/powerfist/proc/updateTank(obj/item/weapon/tank/emergency_oxygen/thetank, removing = 0, mob/living/carbon/human/user)
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/powerfist.dm]: you idiot, Fox, why did you typecast this to only emergency oxygen tanks when it accepts all of them.

**Comment by taukausanake** [comment]
I'm never going to stop asking stupid questions. :) Can I use plasma tanks to unleash fiery punches of justice?

**Comment by Spacemanspark** [comment]
Can I punt you down the hallway with this and not have to hear another question

**Comment by Vivalas** [comment]
Now I have a mental images of swarms of redshirts charging at one dude with a powerfist and being punched to oblivion.

Should make it also make them drop their things. (Make it blow their shoes off for extra fun.)

**Comment by taukausanake** [comment]
That vision makes me want to flurry of punch said swarm slowmo. I REALLY want to see what happens when you use this along with time slow. Do they get punched but keep the energy until time slow ends or are they punched out of the area? SO MANY QUESTIONS, ANSWER THEM SPARK!

**Comment by Spacemanspark** [comment]
:hocho:

**Pull request merged**